airflow/operators/python.py

```diff
@@ -185,16 +186,37 @@ def execute(self, context: Dict):
 class ShortCircuitOperator(PythonOperator, SkipMixin):
     """
     Allows a workflow to continue only if a condition is met. Otherwise, the
-    workflow "short-circuits" and downstream tasks are skipped.
+    workflow "short-circuits" and downstream tasks are skipped. The short-circuiting can be configured to
+    operated in one of two modes: "hard" or "soft".  In "hard" mode, the default mode, all downstream tasks
+    are skipped without considering the ``trigger_rule`` defined for tasks.  In "soft" mode, the
+    ``trigger_rule`` is respected.
 
     The ShortCircuitOperator is derived from the PythonOperator. It evaluates a
-    condition and short-circuits the workflow if the condition is False. Any
-    downstream tasks are marked with a state of "skipped". If the condition is
-    True, downstream tasks proceed as normal.
+    condition and short-circuits the workflow if the condition is False. Any downstream tasks are marked with
+    a state of "skipped" based on the short-circuiting mode configured. If the condition is True, downstream
+    tasks proceed as normal.
 
-    The condition is determined by the result of `python_callable`.
+    The condition is determined by the result of ``python_callable``.
+
+    :param mode: If set to "hard", all downstream tasks from this operator task will be skipped.  This is
+        the default behavior.  If set to "soft", downstream tasks will be skipped but the ``trigger_rule``
+        defined for a task will be respected.
+    :type mode: bool
     """
 
+    modes = ["hard", "soft"]
```

Comment:
I wonder if it would be better to have:
```suggestion
    downstream_trigger_rules_mode = ["respect", "ignore"]
```
At least for me i can understand from the parameter name and option what it means without looking it up in the docstring/documentation. WDYT?

Reply:
why not use the same semantics than for the sensors ( soft/hard )->

https://github.com/apache/airflow/pull/8867

soft_skip:bool = False

Reply:
@josh-fell ?

Reply:
Looks mich better now indeed Hard/Soft was extremely difficult to mentally map what it could mean.